files/justfiles/secureblue.just

```diff
@@ -682,3 +682,329 @@ debug-info:
+            esac
+            resolver_https_address="https://$resolver_hostname/dns-query"
+            ;;
+        5)
+            echo "Setting custom resolver."
+            echo "NOTE: If the resolver does not support DoT/DoQ or DNSSEC, this process will not work."
+            echo ""
+            echo "Please provide the technical information."
+            read -p "Please enter the resolver's IP address (e.g. '1.1.1.2'): " resolver_ipv4_address
+            read -p "Does the resolver provide two distinct IP addresses (e.g. '1.1.1.2' and '1.0.0.2')? [Y/n] " resolver_has_second_ip
+            if [[ "$resolver_has_second_ip" == [Yy]* ]]; then
+                read -p "Please enter the resolver's second IP address: " resolver_ipv4_address_2
+            fi
+            read -p "Does the resolver support IPv6 (e.g. '2606:4700:4700::1112')? [Y/n] " resolver_supports_ipv6
+            if [[ "$resolver_supports_ipv6" == [Yy]* ]]; then
+                read -p "Please enter the resolver's IPv6 address: " resolver_ipv6_address
+                if [[ "$resolver_has_second_ip" == [Yy]* ]]; then
+                    read -p "Please enter the resolver's second IPv6 address: " resolver_ipv6_address_2
+                fi
+            fi
+            read -p "Please enter the second resolver's hostname (e.g. 'security.cloudflare-dns.com'): " resolver_hostname
+            ;;
+    esac
+
+    if [[ "$resolver_selection" == 0 ]]; then
+        if [[ -f "$policy_file" ]]; then
+            rm $policy_file
+            echo "Removed browser policy."
+        fi
+        cat << EOF > "$resolved_conf"
+[Resolve]
+DNSSEC=allow-downgrade
+DNSOverTLS=opportunistic
+EOF
+        systemctl restart systemd-resolved
+        echo "Configuration file for resolved reset and service restarted."
+        exit 0
+    fi
+
+    read -p "Would you like the resolver to be set in the default browser (hardened-chromium) via management policy? [Y/n] " set_browser_policy
+    if [[ "$set_browser_policy" == [Yy]* && "$resolver_selection" == 5 ]]; then
+        read -p "Please enter the second resolver's HTTPS address (e.g. 'https://security.cloudflare-dns.com/dns-query'): " resolver_https_address
+    fi
+
+    resolved_conf_dns_string="DNS="
+    resolved_conf_dns_string+=" $resolver_ipv4_address"
+    resolved_conf_dns_string+="#$resolver_hostname"
+    if [[ "$resolver_has_second_ip" == [Yy]* ]]; then
+        resolved_conf_dns_string+=" $resolver_ipv4_address_2"
+        resolved_conf_dns_string+="#$resolver_hostname"
+    fi
+    if [[ "$resolver_supports_ipv6" == [Yy]* ]]; then
+        resolved_conf_dns_string+=" $resolver_ipv6_address"
+        resolved_conf_dns_string+="#$resolver_hostname"
+        if [[ "$resolver_has_second_ip" == [Yy]* ]]; then
+            resolved_conf_dns_string+=" $resolver_ipv6_address_2"
+            resolved_conf_dns_string+="#$resolver_hostname"
+        fi
+    fi
+
+    cat << EOF > "$resolved_conf"
+[Resolve]
+DNSSEC=true
+DNSOverTLS=true
```

Comment:
Does this mean that, if fully configured, chromium's dns requests will be sent via dnsoverhttps, but other dns requests will go via DoT?

In that case why not just leave chromium in the default state and let all requests go over DoT?

Reply:
Also, we should put a check for

```
DNSSEC=true
DNSOverTLS=true
```

in the audit script